ED/protocol.html

```diff
@@ -666,30 +666,26 @@ <h2 property="schema:name">Hypertext Transfer Protocol</h2>
+                  <p><span about="" id="server-unauthenticated" rel="spec:requirement" resource="#server-unauthenticated"><span property="spec:statement">When a client does not provide valid credentials when requesting a resource that requires it (see <a href="#webid">WebID</a>), <span rel="spec:requirementSubject" resource="#Server">servers</span> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> send a response with a <code>401</code> status code (unless <code>404</code> is preferred for security reasons).</span></span></p>
 
-                  <p><span about="" id="server-authentication" rel="spec:requirement" resource="#server-authentication"><span property="spec:statement"><span rel="spec:requirementSubject" resource="#Server">Servers</span> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> conform to <cite>HTTP/1.1 Authentication</cite> [<cite><a class="bibref" href="#bib-rfc7235">RFC7235</a></cite>].</span></span> <span about="" id="server-unauthenticated" rel="spec:requirement" resource="#server-unauthenticated"><span property="spec:statement">When a client does not provide valid credentials when requesting a resource that requires it (see <a href="#webid">WebID</a>), <span rel="spec:requirementSubject" resource="#Server">servers</span> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> send a response with a <code>401</code> status code (unless <code>404</code> is preferred for security reasons).</span></span></p>
+                  <p><span about="" id="server-content-type" rel="spec:requirement" resource="#server-content-type"><span property="spec:statement"><span rel="spec:requirementSubject" resource="#Server">Server</span> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> generate a <code>Content-Type</code> header field in a message that contains content.</span></span></p>
```

Comment:
```suggestion
                  <p><span about="" id="server-content-type" rel="spec:requirement" resource="#server-content-type"><span property="spec:statement"><span rel="spec:requirementSubject" resource="#Server">Server</span> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> generate a <code>Content-Type</code> header field in a message with content.</span></span></p>
```

Based on this comment: https://github.com/solid/specification/pull/524#issuecomment-1544288113

As I read it, the message has content, but a response to HEAD never *includes* content. I'm afraid that *contains* could be confused with *includes*, leading to the interpretation that this requirement doesn't apply to HEAD.

Reply:
Good catch!

I'd prefer to correct the term itself though, rather than the phrasing, since what we are really talking about in that statement is a representation (representation data/content + representation metadata), not content.

Simple replacement:

```suggestion
                  <p><span about="" id="server-content-type" rel="spec:requirement" resource="#server-content-type"><span property="spec:statement"><span rel="spec:requirementSubject" resource="#Server">Server</span> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> generate a <code>Content-Type</code> header field in a message that contains a representation.</span></span></p>
```

Replacement with additional explicit mention of HEAD.

```suggestion
                  <p><span about="" id="server-content-type" rel="spec:requirement" resource="#server-content-type"><span property="spec:statement"><span rel="spec:requirementSubject" resource="#Server">Server</span> <span rel="spec:requirementLevel" resource="spec:MUST">MUST</span> generate a <code>Content-Type</code> header field in a message that contains a representation, even if that representation only consistss of representation metadata, as in a `HEAD` request.</span></span></p>
```

Reply:
Based on https://datatracker.ietf.org/doc/html/rfc9110#name-representations (added bold)
> A [target resource](https://datatracker.ietf.org/doc/html/rfc9110#target.resource) might be provided with, or be capable of generating, multiple representations that are each intended to reflect the resource's current state. An algorithm, usually based on [content negotiation](https://datatracker.ietf.org/doc/html/rfc9110#content.negotiation) ([Section 12](https://datatracker.ietf.org/doc/html/rfc9110#content.negotiation)), would be used to select one of those representations as being most applicable to a given request. **This "selected representation" provides the data and metadata for evaluating conditional requests ([Section 13](https://datatracker.ietf.org/doc/html/rfc9110#conditional.requests)) and constructing the content** for [200 (OK)](https://datatracker.ietf.org/doc/html/rfc9110#status.200), [206 (Partial Content)](https://datatracker.ietf.org/doc/html/rfc9110#status.206), and [304 (Not Modified)](https://datatracker.ietf.org/doc/html/rfc9110#status.304) responses to GET ([Section 9.3.1](https://datatracker.ietf.org/doc/html/rfc9110#GET)).

The message has content, this content gets included for `GET` and doesn't get included for `HEAD`.
Since we talk about the message, not the resource, we should stick to content and not representation.

I like mentioning that this applies to HEAD, just to be crystal clear, we also have the option to do it by adding a note under the requirement linking to https://datatracker.ietf.org/doc/html/rfc9110#section-6.4.1

Reply:
'Representation' is a term also used separately from 'selected representation'. In most of the method definitions, for example, the RFC talks about representations as part of a request or response.

We can tackle that in a separate PR, but should we then revert the last commit here, or do we temporarily live with the semantically incorrect sentence?

Reply:
'Representation' is a term also used separately from 'selected representation', as that what is transferred in a message:

> A "representation" is information that is intended to reflect a **past, current, or desired state** of a given resource, in a format that can be readily **communicated via the protocol** [3.2] ... A server responds to a client's request by sending one or more "response" messages, each including a status code ... resource metadata, and representation metadata, content [etc.] [3.4] ... **HTTP messages often transfer a complete or partial representation as the message "content"** [6.4] ... The representation data associated with an HTTP message is either **provided as the content of the message** or referred to by the message semantics and the target URI. The representation data is in a format and encoding defined by the representation metadata header fields. **The data type of the representation data is determined via the header fields Content-Type** and Content-Encoding [8.1] ... Representation header fields provide metadata about the representation. When a message includes content, the representation header fields describe how to interpret that data. **In a response to a HEAD request, the representation header fields describe the representation data that would have been enclosed in the content if the same request had been a GET** [8.2] ... The "Content-Type" header field indicates the media type of the associated representation: either the **representation enclosed in the message content** or the selected representation, as determined by the message semantics [8.3] ... HEAD is used to obtain metadata about the selected representation without transferring its representation data [9.3.2]

We can tackle that in a separate PR, but should we then revert the last commit here, or do we temporarily live with the semantically incorrect sentence?

Reply:
Let's track it with https://github.com/solid/specification/issues/565
IMO we can live with what you currently have there and make sure to resolve the dedicated issue before releasing the next WD.